src/test/regress/upgrade/custom_citus.py

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env python3
+
+"""custom_citus
+Usage:
+    custom_citus --bindir=<bindir> --pgxsdir=<pgxsdir> --parallel=<parallel>
+
+Options:
+    --bindir=<bindir>              The PostgreSQL executable directory(ex: '~/.pgenv/pgsql-11.3/bin')
+    --pgxsdir=<pgxsdir>           	       Path to the PGXS directory(ex: ~/.pgenv/src/postgresql-11.3)
+    --parallel=<parallel>           how many configs to run in parallel
+"""
+
+import upgrade_common as common
+import threading
+import atexit
+from docopt import docopt
+import os, shutil
+import time
+import sys
+
+from config import (
```

Comment:
I have been looking into different configurations we can use, and here are my suggestions (open for discussion):
- Always MX
- Always coordinator is added to the metadata

Should keep:
- CitusSingleNodeClusterConfig
- CitusSingleWorkerClusterConfig
- CitusSuperUserDefaultClusterConfig

New ones:
- CitusFiveWorkersManyShardsClusterConfig: SET citus.shard_count TO 494;
- CitusSmallSharedPoolSizeConfig: SET citus.max_shared_pool_size TO 2; SET citus.local_shared_pool_size TO 2
- CitusSmallExecutorPoolSizeConfig: SET citus.max_adaptive_executor_pool_size TO 2;
- CitusSequentialExecutionConfig: SET citus.multi_shard_modify_mode TO sequential;
- CitusCacheManyConnectionsConfig: SET citus.max_cached_conns_per_worker TO 4;
- CitusUnusualExecutorConfig: SET citus.max_adaptive_executor_pool_size TO 7; SET citus.executor_slow_start_interval TO 1; SET citus.prevent_incomplete_connection_establishment TO false && citus.enable_cost_based_connection_establishment TO false; SET citus.max_cached_connection_lifetime TO 10ms; SET citus.force_max_query_parallelization TO on; SET citus.binary_worker_copy_format TO false; SET citus.enable_binary_protocol TO false;
- CitusSmallCopyBuffers: SET citus.copy_switchover_threshold TO '1B';  SET citus.local_copy_flush_threshold TO '1B'; SET citus.remote_copy_flush_threshold TO '1B';
- CitusUnusualQuerySettings:Set citus.task_assignment_policy to "first-replica" ; SET citus.enable_fast_path_router_planner TO false; SET citus.enable_local_execution TO false;  SET citus.enable_single_hash_repartition_joins TO true; SET citus.recover_2pc_interval TO '1s';  SET citus.remote_task_check_interval TO '1ms';


Some of the combinations might fail for some queries, but we can tweak that later.

Can you apply these configurations in the PR while I work for adding queries?

Reply:
1 more class would be to have coordinator's shouldhaveshard = True. I guess we do that in `CitusFiveWorkersManyShardsClusterConfig` ?